pybamm/models/submodels/porosity/base_porosity.py

```diff
@@ -16,7 +16,7 @@ class BaseModel(pybamm.BaseSubModel):
     **Extends:** :class:`pybamm.BaseSubModel`
     """
 
-    def __init__(self, param):
+    def __init__(self, param, options):
         super().__init__(param)
```

Comment:
```suggestion
        super().__init__(param, options=options)
```

Reply:
If you pass options as argument, the `base_submodel` will take care of doing `self.options=options` for you so you don't need it in the other classes.